### DISCUSSION TITLE
flutter fcm IOS not working(Resume, launch)

### DISCUSSION THREAD

**Discussion by NakGun** [firebase/flutterfire#4904] [Q&A] [answered]
hi im Korean my English not good sorry understand me

I make chat app so notification is very important

I'm done android fcm working good
 - node.js , make channel
but iOS not working resume, launch only onMessage fire(foreground)
my architecture
  - flutter 1.22.0
  - ios14.4
  - firebase_messaging: 6.0.16 
  - firebase_auth: ^0.15.5+2
  - firebase_core: ^0.4.4+3
  - not using flutter_local_notification
  - flutter doctor is good

- SEND(FLUTTER)
  final HttpsCallable sendFCM = CloudFunctions.instance
      .getHttpsCallable(functionName: 'sendFCM') 
    ..timeout = const Duration(seconds: 30); 

    final HttpsCallableResult result = await sendFCM.call(
      <String, dynamic>{
        "token": setupRCV['token'],
        "title": title,
        "body": body,
        "name" : myData.username,
        "picture" : myData.profileImg1,
        "reqTp" : talkType,
        "roomID" : roomID
      },
    );

- Index.js
const functions = require('firebase-functions');
const admin = require("firebase-admin");

admin.initializeApp();

exports.sendFCM = functions.https.onCall((data, context) => {
  var token = data["token"];
  var title = data["title"];
  var body = data["body"];
  var picture = data["picture"];
  var name = data["name"];
  var reqTp = data["reqTp"];
  var roomID = data["roomID"];

  var payload = {  	
    "notification": {
      "title": "title",
      "body": name,      
      "channel_id" : "........",
      "click_action" : "FLUTTER_NOTIFICATION_CLICK",            
      "default_sound": "false",
      "default_vibrate_timings": "false",      
      "sound": "ringbell",         
    },
    "data" : {
      "title": title,
      "body": body,         
    	"accept":"Accept",
    	"reject":"Reject",
    	"picture":picture,
    	"name":name,
    	"reqTp":reqTp,
    	"roomID":roomID,   
    	"sound": "ringbell", 	  
    	"channel_id" : "......",
    	"click_action" : "FLUTTER_NOTIFICATION_CLICK",    	
    },  
  }

  var options = {
	  priority: 'high',
	  timeToLive: 60,
	};

  var result = admin.messaging().sendToDevice(token, payload, options);
  return result;
})

- info.plist
    <key>FirebaseScreenReportingEnabled</key>
    <true/>
    <key>FirebaseAppDelegateProxyEnabled</key>
    <true/>

- AppDelegate
     if #available(iOS 10.0, *) {
      UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
    }
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
    @available(iOS 10.0, *)
    override func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
        completionHandler([.alert,.sound,.badge])
    }

- main
- init() {
    if (Platform.isIOS) {
      _messaging.requestNotificationPermissions(
          IosNotificationSettings(sound: true, badge: true, alert: true, provisional: false)
      );
      _messaging.onIosSettingsRegistered.listen((IosNotificationSettings settings) {
        print("Settings registered: $settings");
      });
    }

    _messaging.getToken().then((token) {
      print('token:${token}');
    });

    _messaging.configure(
      onMessage: (Map<String, dynamic> message) async {
        print('<<<<<<<<<<>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>onMessage');
      },
      onLaunch: (Map<String, dynamic> message) async {
        print('<<<<<<<<<<>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>onLaunch');
      },
      onResume: (Map<String, dynamic> message) async {
        print('<<<<<<<<<<>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>onResume');
      },
      onBackgroundMessage: null,
    );
  }

===============================================================================
I need background and kill status notification on header 
and click resume or launch event fire

foreground onMessage fire

plz help I'm everything try 
but plugin version upgrade solution is can't me
I think maybe node.js problem...but not find solution 
thank you read~~!!

**Comment by Anoirwork** [comment, marked as answer]
Hi there i've been looking all over the internet and i found out that FCM background notification was not supported until the version 8.0.0-dev so you need probably to update your version hope this.
Have a great day, Stay safe.

**Reply by NakGun** [reply]
thank you Anoirwork
my project is old version..so difficult version upgrade haha;
is not good...situation ㅜㅜ
